src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -2,22 +1,48 @@
+import { Dispatch, SetStateAction } from 'react';
+import { Filter } from '../../App';
+import { useDispatch } from 'react-redux';
+import { filterSlice } from '../../features/filter';
+
+type Props = {
+  filterType: string;
+  query: string;
+  onQuery: Dispatch<SetStateAction<string>>;
+  onResetQuery: () => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filterType,
+  query,
+  onQuery,
+  onResetQuery,
+}) => {
+  const dispatch = useDispatch();
 
-export const TodoFilter: React.FC = () => {
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={filterType}
+            onChange={event => {
+              dispatch(filterSlice.actions.toggleFilter(event.target.value));
```

Comment:
Ensure that the `toggleFilter` action in `filterSlice` is correctly defined to handle the filter type changes. This action should update the state based on the selected filter type.